api/types/src/transaction.rs

```diff
@@ -1228,6 +1235,89 @@ impl
     }
 }
 
+/// Fee payer signature for fee payer transactions
+///
+/// This allows you to have transactions across multiple accounts and with a fee payer
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Object)]
+pub struct FeePayerSignature {
+    pub sender: AccountSignature,
+    /// The other involved parties' addresses
+    pub secondary_signer_addresses: Vec<Address>,
+    /// The associated signatures, in the same order as the secondary addresses
+    pub secondary_signers: Vec<AccountSignature>,
+    /// The address of the paying party
+    pub fee_payer_address: Address,
+    /// The signature of the fee payer
+    pub fee_payer_signer: AccountSignature,
+}
+
+impl VerifyInput for FeePayerSignature {
+    fn verify(&self) -> anyhow::Result<()> {
+        self.sender.verify()?;
+
+        for signer in self.secondary_signers.iter() {
+            signer.verify()?;
+        }
+        self.fee_payer_signer.verify()?;
```

Comment:
Do we need to verify that this signature matches the fee payer address here? Should at least be done in the execution flow somewhere

Reply:
It's done in the prologue

Reply:
Found this in Move in fee_payer_script_prologue. I wonder if we should have a separate error here because it'd be hard to know which signature mismatches with the address - senders' or payer's. Users would see the same error for both cases and be confused